Objects/exceptions.c

```diff
@@ -2862,6 +2862,40 @@ _PyErr_TrySetFromCause(const char *format, ...)
  * or minus, using the stream redirection syntax).
  */
 
+
+// Static helper for setting legacy print error message
+static int
+_set_legacy_print_statement_msg(PySyntaxErrorObject *self, Py_ssize_t start)
+{
+    /*
+    `data` is UTF-8 encoded and treated as an indivisible opaque blob. All
+    we know about it is that it starts with `print ` and may end with a `,`
+    in which case we also append suggest_end_arg which makes the input
+    analogous to Python3 print syntax.
+    */
+
+    PyObject *strip_sep_obj = PyUnicode_FromString("\r\n");
+    PyObject *data = _PyUnicode_XStrip(self->text, 1, strip_sep_obj);
+    // PRINT_OFFSET is to remove print word from the data.
+    const int PRINT_OFFSET = 6;
+    Py_ssize_t text_len = PyUnicode_GET_LENGTH(data);
+    data = PyUnicode_Substring(data, PRINT_OFFSET, text_len);
+    // gets the modified text_len after stripping `print `
+    text_len = PyUnicode_GET_LENGTH(data);
+    char *maybe_end_arg = " end=\" \"";
+
+    PyObject *error_msg = PyUnicode_FromFormat(
+        "Missing parentheses in call to 'print'. Did you mean print(%U)?",
+         data);
+    if (PyUnicode_Tailmatch(data, PyUnicode_FromString(","), start, text_len, 1))
+        error_msg = PyUnicode_FromFormat(
```

Comment:
This is almost duplicates the former formatting. Both cases can be merged if use an empty string for `maybe_end_arg` in the first case.